dali/pipeline/operator/builtin/make_contiguous.cu

```diff
@@ -16,12 +16,68 @@
 
 namespace dali {
 
-DALI_REGISTER_OPERATOR(MakeContiguous, MakeContiguous, Mixed);
+void MakeContiguousMixed::Run(MixedWorkspace &ws) {
+  const auto& input = ws.Input<CPUBackend>(0, 0);
+  int sample_dim = input.shape().sample_dim();
+  TensorListShape<> output_shape(batch_size_, sample_dim);
+  TensorLayout layout = input.GetLayout();
+  TypeInfo type = input.type();
+  size_t total_bytes = 0;
+  for (int i = 0; i < batch_size_; ++i) {
+    auto &sample = ws.Input<CPUBackend>(0, i);
+    output_shape.set_tensor_shape(i, sample.shape());
+    size_t sample_bytes = sample.nbytes();
+    if (coalesced && sample_bytes > COALESCE_TRESHOLD)
+      coalesced = false;
+    total_bytes += sample_bytes;
+    DALI_ENFORCE(type == sample.type(), "Inconsistent types in "
+        "input batch. Cannot copy to contiguous device buffer.");
+    DALI_ENFORCE(sample_dim == sample.shape().sample_dim(), "Inconsistent sample dimensions "
+        "in input batch. Cannot copy to contiguous device buffer.");
+  }
 
-DALI_SCHEMA(MakeContiguous)
-  .DocStr(R"code(Move input batch to a contiguous representation, more suitable for execution on the GPU)code")
-  .NumInput(1)
-  .NumOutput(1)
-  .MakeInternal();
+  auto &output = ws.Output<GPUBackend>(0);
+  output.Resize(output_shape);
+  output.SetLayout(layout);
+  output.set_type(type);
+
+  if (coalesced) {
+    TimeRange tm("coalesced", TimeRange::kBlue);
+
+    if (!cpu_output_buff.capacity()) {
+      size_t alloc_size = std::max<size_t>(total_bytes, batch_size_*bytes_per_sample_hint);
```

Comment:
I think reserving `batch_size_ * bytes_per_sample_hint` could be moved to ctor

Reply:
No, if we get rid of batch_size_ .

Reply:
But I see it is not used anyway.